src/core/src/service_interfaces/LifecycleManagerArc.py

```diff
@@ -65,12 +65,12 @@ def execution_start_check(self):
 
             # attempted sequence number is same as recorded core sequence - expected
             if int(self.execution_config.sequence_number) == int(core_sequence['number']):
-                if bool(core_sequence['completed']):
+                if core_sequence['completed'].lower() == 'false':
```

Comment:
Looks like logic is wrong here. it should be core_sequence['completed'].lower() == 'true'  -> which means previous operation on the same sequence number is completed and current autoassessment can proceed ( safe to start ) else Its not safe to start auto assessment

Reply:
same if statement is translated properly in LifecycleManagerAzure.py  line number 58

Reply:
Changed. This is a good example of the importance of having unit tests along with the code change. I caught this issue from a UT for Azure and missed adding the same in Arc. A UT in Arc would have caught this.